app/routes.js

```diff
@@ -1,60 +1,25 @@
-const { check } = require('express-validator/check');
 const userController = require('./controllers/users');
 const albumController = require('./controllers/albums');
-const { tokenCheck } = require('./middlewares/validations');
+const {
+  tokenCheck,
+  paramValidation,
+  userParamsValidations,
+  newSessionParamsValidation,
+  checkAdmin
+} = require('./middlewares/validations');
 
 exports.init = app => {
-  // app.get('/endpoint/get/path', [], controller.methodGET);
-  // app.put('/endpoint/put/path', [], controller.methodPUT);
-  // app.post('/endpoint/post/path', [], controller.methodPOST);
-  const userValidations = [
-    check('name')
-      .isLength({ min: 1 })
-      .withMessage('Name Required'),
-    check('surname')
-      .isLength({ min: 1 })
-      .withMessage('Surname Required'),
-    check('password')
-      .isLength({ min: 8 })
-      .withMessage('Password must contain more than 8 characters.'),
-    check('email')
-      .isEmail()
-      .withMessage('Invalid email.')
-      .custom(email => {
-        if (!email.endsWith('wolox.com.ar')) {
-          throw new Error('Email is not from wolox domain!');
-        }
-        return true;
-      })
-  ];
-
-  app.post('/users', userValidations, userController.userCreate);
-
-  app.post(
-    '/users/sessions',
-    [
-      check('password')
-        .isLength({ min: 8 })
-        .withMessage('Password must contain more than 8 characters.'),
-      check('email')
-        .isEmail()
-        .withMessage('Invalid email.')
-        .custom(email => {
-          if (!email.endsWith('wolox.com.ar')) {
-            throw new Error('Email is not from wolox domain!');
-          }
-          return true;
-        })
-    ],
-    userController.userNewSession
-  );
-
+  app.post('/users', userParamsValidations, userController.userCreate);
+  app.post('/users/sessions', newSessionParamsValidation, userController.userNewSession);
   app.post('/users/sessions/invalidate_all', [tokenCheck], userController.invalidateAllSessions);
   app.get('/users', [tokenCheck], userController.usersList);
   app.get('/users/:id/albums', [tokenCheck], userController.albumList);
   app.get('/users/albums/:id/photos', [tokenCheck], userController.albumPhotosList);
-  app.post('/admin/users', [tokenCheck].concat(userValidations), userController.userAdminCreate);
-
+  app.post(
+    '/admin/users',
+    [tokenCheck, checkAdmin].concat(userParamsValidations),
```

Comment:
if you add it as a normal array element like `[tokenCheck, checkAdmin, userParamsValidations]` doesn't it work? As far as I remember express doesn't mind array of array of middlewares

Reply:
Fixed